src/librustc_typeck/check/compare_method.rs

```diff
@@ -475,7 +483,10 @@ pub fn compare_const_impl<'tcx>(tcx: &ty::ctxt<'tcx>,
     });
 
     match err {
-        Ok(()) => { }
+        Ok(RelateOk { obligations, .. }) =>
+            for obligation in obligations {
+                fulfillment_cx.register_predicate_obligation(&infcx, obligation);
```

Comment:
These obligations are registered, but the fulfillment context is never drained (nor was it drained earlier in history just prior to these commits). Is it supposed to get drained?